ca/config.go

```diff
@@ -148,10 +154,36 @@ func NewConfigPaths(baseCertDir string) *SecurityConfigPaths {
 	}
 }
 
+// GenerateJoinToken creates a new join token.
+func GenerateJoinToken(rootCA *RootCA) string {
+	var secretBytes [generatedSecretEntropyBytes]byte
+
+	if _, err := cryptorand.Read(secretBytes[:]); err != nil {
+		panic(fmt.Errorf("failed to read random bytes: %v", err))
+	}
+
+	var nn, digest big.Int
+	nn.SetBytes(secretBytes[:])
+	digest.SetString(rootCA.Digest.Hex(), 16)
+	return fmt.Sprintf("SWMTKN-1-%s-%0[2]*s", digest.Text(joinTokenBase), maxGeneratedSecretLength, nn.Text(joinTokenBase))
```

Comment:
We're zero-padding the secret to 128 bits. It's not really necessary, but I felt like this would be a good idea so the length of the secret doesn't leak the upper bits. Happy to change it.